plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -134,65 +134,65 @@ async fn run(
+
+    loop {
+        tokio::select! {
+            message = mqtt_client.received.next() => {
+                if let Some(message) = message {
+                    debug!("Received {:?}", message);
+                    if let Ok(payload) = message.payload_str() {
+                            let result = match payload.split(',').next().unwrap_or_default() {
+                                "524" => {
+                                    let config_download_request =
+                                        SmartRestConfigDownloadRequest::from_smartrest(payload)?;
+
+                                    handle_config_download_request(
+                                        &plugin_config,
+                                        config_download_request,
+                                        tmp_dir.clone(),
+                                        mqtt_client,
+                                        http_client,
+                                    )
+                                    .await
+                                }
+                                "526" => {
+                                    // retrieve config file upload smartrest request from payload
+                                    let config_upload_request =
+                                        SmartRestConfigUploadRequest::from_smartrest(payload)?;
+
+                                    // handle the config file upload request
+                                    handle_config_upload_request(
+                                        &plugin_config,
+                                        config_upload_request,
+                                        mqtt_client,
+                                        http_client,
+                                    )
+                                    .await
+                                }
+                                _ => {
+                                    // Ignore operation messages not meant for this plugin
+                                    Ok(())
+                                }
+                            };
+                         if let Err(err) = result {
+                            error!("Handling of operation: '{payload}' failed with {err}");
+                        }}
+                    } else {
+                        // message is None and the connection has been closed
+                        return Ok(());
                     }
+            }
+        Some(Ok(event)) = inotify_stream.next() => {
+            match event.mask {
+                EventMask::CLOSE_WRITE => {
+                    plugin_config = handle_dynamic_config_update(mqtt_client, config_file_path).await?;
                 }
-            };
-
-            if let Err(err) = result {
-                error!("Handling of operation: '{payload}' failed with {err}");
+                _ => {}
             }
         }
+        }
     }
-
-    mqtt_client.close().await;
-
-    Ok(())
```

Comment:
Why these two lines were removed?

Reply:
I believe this is addressed in the [else above](https://github.com/thin-edge/thin-edge.io/pull/1187/files/929f1d95970b050f4b8232f162dcf8f624bd08a1#r897962089), no?